### DISCUSSION TITLE
Is the "Grove - Gas Sensor V2(Multichannel)" supported

### DISCUSSION THREAD

**Discussion by ChrisROL** [arendst/Tasmota#20305] [Ideas]
Hi,

i try to find out, if the second version of the grove multichannel gas sensor is already supported and i just can't find the correct setting.
The first version is supported by switch USE_MGS, but it has different address and sensors to the V2

https://wiki.seeedstudio.com/Grove-Multichannel-Gas-Sensor-V2/

https://wiki.seeedstudio.com/Grove-Multichannel_Gas_Sensor/

Regards
Christian

**Comment by Jason2866** [comment]
No, it is not supported.

**Comment by ChrisROL** [comment]
Is there a posibility to integrate it also, or i am the only one who wants to use it?

**Reply by barbudor** [reply]
so far you are the 1st one
If you are using an ESP32, it should be fairly simple to recode the driver in Berry

**Reply by ChrisROL** [reply]
I want to use it with a nodemcu and i haven't done any programming in berry.
My professional is  programming on siemens PLC and i'm not good in higher level programming language.
The scripting in tasmota is ok for me

**Comment by Jason2866** [comment]
If "someone" who has the device writes a driver for Tasmota.

**Reply by sfromis** [reply]
If it is "fairly similar" to the older generation, it might be feasible to extend the same driver to handle both.

**Reply by ChrisROL** [reply]
I have only seen a library on the manufactur's page. 
One for the old and one for v2.

**Comment by spockatieee** [comment]
Any updates on the v2 version of the Grove Gas sensor for Tasmota? I've tried to tinker with the v1 code a bit, but no success. 

I do see that the Gas Sensor v2 is supported in esphome. Would it be possible to port that over? Here is the link to their driver:

https://github.com/esphome/esphome/tree/dev/esphome/components/grove_gas_mc_v2
